src/utilities/utilities.jl

```diff
@@ -421,7 +421,45 @@ const julia_remote = Remotes.GitHub("JuliaLang", "julia")
 """
 Stores the memoized results of [`getremote`](@ref).
 """
-const GIT_REMOTE_CACHE = Dict{String,Union{Remotes.Remote,Nothing}}()
+const GIT_REMOTENAME_CACHE = Dict{String, String}()
+const GIT_REMOTEURL_CACHE = Dict{String,Union{Remotes.Remote,Nothing}}()
+
+"""
+$(TYPEDSIGNATURES)
+
+Determines the GitHub remote of a directory, returninf 'origin' if that remote  exists,
+and otherwise returning the result of `git config --get remote.pushDefault`.
+If neither exist, throw an error.
+
+The results for a given directory are memoized in [`GIT_REMOTENAME_CACHE`](@ref), since calling
+`git` is expensive and it is often called on the same directory over and over again.
+"""
+function getremotename(dir::AbstractString)
+    isdir(dir) || return nothing
+    return get!(GIT_REMOTENAME_CACHE, dir) do
+        try
+            originurl = readchomp(setenv(`$(git()) config --get remote.origin.url`; dir=dir))
+            @assert !isempty(originurl)
+            return "origin"
+        catch; end
```

Comment:
You seem to be using `@assert` plus `try/catch` for control flow. That seems like a bad idea for multiple reasons (e.g. future Julia version might allow disabling `@assert` checks, which is, I think, mainly intended to enforce to check runtime invariants, which is not what this code does).

How about just e.g. this
```suggestion
        originurl = readchomp(setenv(`$(git()) config --get remote.origin.url`; dir=dir))
        if !isempty(originurl)
            return "origin"
        end
```

Reply:
Same applies to the rest of the code, of course